prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -56,23 +86,338 @@ def _retrieve_redfish_sensor_data(self) -> List[Any]:
         Returns:
             sensors: List of dicts with details for each sensor
         """
-        try:
-            with redfish.redfish_client(
-                base_url=self.host,
-                username=self.username,
-                password=self.password,
-                timeout=self.timeout,
-                max_retry=self.max_retry,
-            ) as redfish_obj:
-                return redfish_utilities.get_sensors(redfish_obj)
-        except (
-            InvalidCredentialsError,
-            SessionCreationError,
-            ConnectionError,
-            RetriesExhaustedError,
-        ) as err:
-            logger.exception(err)
-        return []
+        sensors: List[Any] = []
+
+        redfish_obj: Optional[HttpClient] = self._get_redfish_obj()
+        if redfish_obj:
+            logger.info("Getting redfish sensor info...")
+            sensors = redfish_utilities.get_sensors(redfish_obj)
+            redfish_obj.logout()
+        return sensors
+
+    def _verify_redfish_call(self, redfish_obj: HttpClient, uri: str) -> Optional[Dict[str, Any]]:
+        """Return REST response for GET request API call on the URI.
+
+        Returns None if URI isn't present.
+        """
+        resp: RestResponse = redfish_obj.get(uri)
+        if resp.status == 200:
```

Comment:
**praise (non-blocking):** 

Instead of using `200`, can we use some `HttpStatusXXX` variable from standard library?

Reply:
After looking into this for a bit, I have a different opinion now.

On checking code in the redfish library, it looks like they are using the status codes directly without `HTTPStatus` as you can see [here](https://github.com/DMTF/python-redfish-library/blob/main/src/redfish/rest/v1.py#L1004) and in the rest of the module. I feel like we can follow the same approach.

Reply:
> It's not different to use 200 or HTTPStatus. Just because HTTPStatus is more clear. https://docs.python.org/3/library/http.html

IMO 200 is already quite clear and it's even used in the main redfish library code. So, we could also follow that leave it at 200 since anyone can infer from the context about the meaning. 

But if you insist, I can change 200 --> `HTTPStatus.OK` which would be more verbose.